### PR TITLE
[semver:patch] Fix changelog - Fail gracefully when no previous deployment found

### PR DESCRIPTION
**Pull request by mattops**
### Checklist

- [ ] All new jobs, commands, executors, parameters have descriptions
- [ ] Examples have been added for any significant new features
- [ ] README has been updated, if necessary

### CODE REVIEW

File: src/jobs/deploy_env.yml [hunk 1]
```diff
@@ -42,10 +42,6 @@ parameters:
     type: string
     default: PROJECT_NAME
     description: The Deployment resource's name in Kubernetes to interrogate for the previous deployment's version
-  fixed_previous_version:
-    type: string
-    default: ""
-    description: (Only used for orb integration testing) Instead of reading from Kubernetes deployment, provide a fixed APP_VERSION for the previous deployment
```
Comment (mikehalmamoj) [inline src/jobs/deploy_env.yml:48]: Apparently this was needed for some testing? https://github.com/ministryofjustice/hmpps-circleci-orb/pull/52#discussion_r658587654

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/jobs/deploy_env.yml (hunk 1 above)
Comment (mattops) [reply src/jobs/deploy_env.yml:48]: no longer - I've just made the changelog script play nice if there is no existing deployment found.

File: src/scripts/version_history.sh [hunk 2]
```diff
@@ -3,20 +3,32 @@
...
+current_commit="$(echo "${APP_VERSION}" | cut -d'.' -f3)"
+
+# Try and get the currently deployed version
+K8S_PREVIOUS_APP_VERSION="$(kubectl get "deployment/${K8S_DEPLOYMENT_NAME}" --namespace="${KUBE_ENV_NAMESPACE}" -o=jsonpath='{.metadata.labels.app\.kubernetes\.io/version}' || true)"
+
 if [ "$K8S_PREVIOUS_APP_VERSION" == "" ]; then
-  K8S_PREVIOUS_APP_VERSION="$(kubectl get "deployment/$K8S_DEPLOYMENT_NAME" --namespace="$KUBE_ENV_NAMESPACE" -o=jsonpath='{.metadata.labels.app\.kubernetes\.io/version}')"
+  # if no previous version was found, set to current commit minus 1
+  echo "Previous deployment not found, showing current commit only." >> .deployment_changelog
+  previous_commit="${current_commit}^1"
```
Comment (sldblog) [inline src/scripts/version_history.sh:18]: 📝 this works on CircleCI, just want to flag it depends on the CI if this history is available
eg. github actions don't pull down anything apart from the exact commit

Comment (sldblog) [reply src/scripts/version_history.sh:18]: To me this is an message case, as the deployment notification already contains the currently deployed SHA

**Pull request merged**